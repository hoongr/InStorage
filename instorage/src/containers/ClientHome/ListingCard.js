import React, { useState, useEffect } from 'react';
import {
  Container,
  Row,
  Col,
  Image,
  Card,
} from 'react-bootstrap';
import { withRouter } from 'react-router-dom';

const ListingCard = (props) => {
  const { listing } = props;
  console.log('listing', listing);
  const {
    _hostAddress,
    _dateEnd,
    _dateStart,
    _inventoryPhotoURL,
    _hostUserID,
  } = listing;

  function formatDate(isoDate) {
    const date = new Date(isoDate);

    return `${date.getMonth()+1}/${date.getDate()+1}`
  }

  function formatAddress(address) {
    return address.split(",")[0]
  }

  return (
  	<div className="pb-5">
	  <style type="text/css">
	    {`
	      .profile-btn {
	        background-color: #7CC6FE;
	        color: white;
	        width: 130px;
	      }
	    `}
      </style>
	  <Card 
	    style={{ 
	      background: "#274060",
	      border: "none",
	    }}
	  >
	    <Card.Img style={{height: "300px", objectFit: "cover"}} variant="top" src={_inventoryPhotoURL} />
	    <Card.Body>
	      <Card.Text >
	        <div>
	      	  <span
		        style={{
		    	  color: "white",
		          fontSize: "18px",
		      	}}
		      >
	            {formatAddress(_hostAddress)}
	          </span>
	          <span className="float-right">
		        <button className="btn profile-btn btn-md" onClick={() => {props.history.push('/user/' + _hostUserID)}}>
			      Contact Host
			    </button>
			  </span>
	        </div>
	        <div
	          style={{
	      	    color: "white",
	      	    fontSize: "14px",
	      	  }}
	      	>
	          {formatDate(_dateStart)} - {formatDate(_dateEnd)}
	        </div>
	      </Card.Text>
	    </Card.Body>
	  </Card>
  	</div>
  )
}

export default withRouter(ListingCard)
