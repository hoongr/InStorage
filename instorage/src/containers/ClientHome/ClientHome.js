import React, { useState, useEffect } from 'react';
import { withRouter } from 'react-router';
import {
  Container,
  Button,
  Row,
  Col,
} from 'react-bootstrap';

import { withAuthorization } from '../../Session';
import ListingCard from './ListingCard';
import FirebaseManager from "../../api/routes/firebaseManager";


const ClientHome = ({history, firebase}) => {
  const [listings, setListings] = useState([]);
  const [pendingListings, setPending] = useState([]);
  const [confirmedListings, setConfirmed] = useState([]);

  useEffect(() => {
    const res = FirebaseManager.getClientBookingRequests(firebase.auth.currentUser.uid);

    res.then(listings => {
      setListings(listings)
      setPending(listings.filter(listing => listing._requestState == "pending"))
      setConfirmed(listings.filter(listing => listing._requestState == "accepted"))
    })
  }, []);

  return (
    <Container style={{ marginTop: '50px', marginBottom: '50px' }}>
      <style type="text/css">
        {`
          .btn-a {
            background-color: #FFA69E;
            color: white;
          }

          .btn-a:hover { color: white }
        `}
      </style>
      <h1>My Bookings</h1>
      {(confirmedListings.length > 0)
        ? (
          <Row className="justify-content-md-left">
            {confirmedListings.map(
              (listing) => (
                <Col key={listing._bookingID} style={{ paddingLeft: '0px' }} xs={4}>
                  <ListingCard listing={listing} />
                </Col>
              ))}
          </Row>
        )
        : (
          <Container>
            <Row className="justify-content-md-center">
              <div>You have no current bookings</div>
            </Row>
            <Row className="justify-content-md-center">
              <Button
                style={{ marginTop: '15px' }}
                variant="a"
                onClick={() => history.push('/Dashboard')}
              >
                Browse Listings
              </Button>
            </Row>
          </Container>
        )
      }
      <h1 style={{ marginTop: '50px' }}>Pending Requests</h1>
      {(pendingListings.length > 0)
        ? (
          <Row className="justify-content-md-left">
            {pendingListings.map(
              (listing) => (
                <Col key={listing._bookingID} style={{ paddingLeft: '0px' }} xs={4}>
                  <ListingCard listing={listing} />
                </Col>
              ))}
          </Row>
        )
        : (
          <Container>
            <Row className="justify-content-md-center">
              <div>You have no pending requests</div>
            </Row>
          </Container>
        )
      }
    </Container>
  );
}

const condition = authUser => !!authUser;
export default withAuthorization(condition)(withRouter(ClientHome));
