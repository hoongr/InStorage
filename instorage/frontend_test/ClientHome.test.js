import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import ClientHome from "../src/containers/ClientHome/ClientHome"
import ListingCard from "../src/containers/ClientHome/ListingCard"

describe('ClientHome', () => {
	const wrapper = shallow(<ClientHome />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('ListingCard', () => {
	const data = {
		_hostAddress: "940 Tiverton Ave",
		_dateEnd: "12/31/2019",
		_dateStart: "12/1/2019",
		_inventoryPhotoURL: "",
	};

	const wrapper = shallow(<ListingCard listing={data}/>);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})