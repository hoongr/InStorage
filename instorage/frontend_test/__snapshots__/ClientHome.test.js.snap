// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientHome should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`ListingCard should match the snapshot 1`] = `ShallowWrapper {}`;
